specs/data_structures.md

```diff
@@ -273,15 +273,20 @@ SMTs can further be extended with _compact_ proofs. [Merkle proofs](#verifying-a
 
 For a Merkle branch of height `h`, an `h`-bit value is appended to the proof. The lowest bit corresponds to the sibling of the leaf node, and each higher bit corresponds to the next parent. A value of `1` indicates that the next value in the list of values provided explicitly in the proof should be used, and a value of `0` indicates that the default value should be used.
 
+Finally, the number of hashing operations can be reduced to be logarithmic in the number of non-empty leaves on average. An internal node that is the root of a subtree that contains exactly one non-empty leaf is replaced by that leaf's leaf node.
```

Comment:
Should we also refer to libra's documentation, where both these optimizations are described and illustrated?
https://developers.libra.org/docs/assets/papers/the-libra-blockchain/2020-04-09.pdf (Section 4.3).